File: python/samples/ecosystem_demo.py
import asyncio

from trinsic.proto.services.account.v1 import AccountDetails
from trinsic.proto.services.provider.v1 import (
    CreateEcosystemRequest,
    InviteRequest,
    ParticipantType,
    InvitationStatusRequest,
)
from trinsic.trinsic_service import TrinsicService
from trinsic.trinsic_util import trinsic_config, set_eventloop_policy


async def ecosystem_demo():
    trinsic_service = TrinsicService(server_config=trinsic_config())
    # createEcosystem() {
    actual_create = await trinsic_service.provider.create_ecosystem(
        request=CreateEcosystemRequest(
            description="My ecosystem", uri="https://example.com"
        )
    )
    # }
    assert actual_create.ecosystem is not None
    assert actual_create.ecosystem.id is not None
    assert actual_create.ecosystem.id.startswith("urn:trinsic:ecosystems:")
    print(f"ecosystem id={actual_create.ecosystem.id}")

    # listEcosystems() {
    # }


if __name__ == "__main__":
    set_eventloop_policy()
    asyncio.run(ecosystem_demo())

File: ruby/test/ecosystem_demo.rb
# frozen_string_literal: true

require 'trinsic_services'
require 'services/account_service'
require 'services/provider_service'

def ecosystem_demo_run
  trinsic = Trinsic::TrinsicService.new(Trinsic.trinsic_server)
  account = trinsic.account_service.login_anonymous

  trinsic.auth_token = account

  # test create ecosystem
  actual_create = trinsic.provider_service.create_ecosystem(Trinsic::Provider::CreateEcosystemRequest.new(
                                             description: 'My ecosystem', uri: 'https://example.com'
                                           ))
  raise 'ecosystem should be created' if actual_create.ecosystem.nil?
  raise 'ecosystem id should exist' if actual_create.ecosystem.id.nil?
  return if actual_create.ecosystem.id.start_with?('urn:trinsic:ecosystems:')

  raise "expected urn prefix='urn:trinsic:ecosystems:'"
end
ecosystem_demo_run
